### ISSUE TITLE
Add more integrations with networks and services

### ISSUE THREAD

**Issue by xhulz** [trufflesuite/vscode-ext#79]
We need to start thinking about an architecture that allows us to add more integrations with networks (L1  & L2) and services.

For instace:
 - IPFS
 - Filecoins
 - ETH2
 - Optimistic

**Comment by michaeljohnbennett** [comment]
The network tree view has the concept of multiple providers to be added to it as treeview items so this code is pretty flexible in this regards.

As to how you work with them its probably still going to be driven from network config in truffle?

**Comment by fainashalts** [comment]
Hi @xhulz are you talking specifically about VSCode here? I'd love to be involved in this discussion and planning when you do it! I think I can help inform what developers will need from the Layer 2 perspective.

**Comment by xhulz** [comment]
Hey @fainashalts, sure thing!

Let's talk to @kevinbluer about that :)

**Comment by kevinbluer** [comment]
Definitely, would be great to brainstorm further @fainashalts. I'll schedule a quick check-in on for later this month. Also for reference I updated the network list to better reflect parity with Infura and the boxes that we have in the pipeline.
